packages/database/test/browser/crawler_support.test.ts

```diff
@@ -38,13 +38,13 @@ describe('Crawler Support', function() {
     restRef = getFreshRepoFromReference(normalRef);
     forceRestClient(false);
 
-    tokenProvider = testAuthTokenProvider(restRef.database.app);
+    // tokenProvider = testAuthTokenProvider(restRef.database.app);
```

Comment:
It seems we are patching auth in app after app.database() has been called.
Under the component framework, authProvider is given as a parameter during database construction, so to mock auth, we should register the mock auth before app.database() is called. It also means we need a new app instance for each test, otherwise database and its authProvider is cached.

My question is how is the mocked auth used in tests. My intention is to ensure the tests with component framework cover all the use cases that are covered today.

I'm also confused because tests pass with these lines commented out.

Reply:
Hm. I don't understand the interactions here well enough to provide an adequate answer (I can do some research if required). If the tests pass against Prod and against the Emulator without mocking the auth token then I see little reason to do so.

Reply:
I double checked all tests (browser, node, emulator) pass without mocking the auth token, so I removed these 2 lines and `testAuthTokenProvider`.